src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,56 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exeptions.UserRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        boolean checkRegistrationAllowance = checkLogin(user.getLogin())
+                && checkPassword(user.getPassword())
+                && checkAge(user.getAge())
+                && !userExists(user);
+        if (!checkRegistrationAllowance) {
+            throw new UserRegistrationException("Registration failed ");
+        }
```

Comment:
Returning a boolean value from those validation methods is not a bad idea but building it into a single check looks hard to read and is actually redundant, this exception here will never be thrown because other ones will be thrown earlier, so I'd rather just call all of the `check...()` methods in order and that's it
BTW you also have no check if the `user` itself is null